src/server/http/xsrf.js

```diff
@@ -4,18 +4,37 @@ export default function (kbnServer, server, config) {
   const disabled = config.get('server.xsrf.disableProtection');
   const versionHeader = 'kbn-version';
   const xsrfHeader = 'kbn-xsrf';
+  const contentTypeHeader = 'content-type';
+  const allowedRequestMediaTypes = ['application/json', 'application/x-ndjson'];
```

Comment:
I thought of several different ways how to handle this:

* Store all allowed media types in array in case we may need to support more content types in the future (current);
* Use single and simple `Regex` expression, but I usually try to avoid `Regex` when it's not really needed;
* Just two additional `content === application/json[x-ndjson]` conditions in `if` statement.

I don't have a strong opinion on what is better, so please let me know what you prefer.